includes/controllers/ApiController.php

```diff
@@ -94,7 +95,148 @@ public function getAllGroups()
+     */
+    public function getAllReactions()
+    {
+        return new ApiResponse($this->getService(ReactionManager::class)->getReactions('', []));
+    }
+
+    /**
+    * @Route("/api/reactions/{id}", methods={"GET"}, options={"acl":{"public"}})
+    */
+    public function getReactions($id)
+    {
+        $id = array_map('trim', explode(',', $id));
+        return new ApiResponse($this->getService(ReactionManager::class)->getReactions('', $id));
+    }
+
+    /**
+     * @Route("/api/user/{userId}/reactions", options={"acl":{"public"}})
+     */
+    public function getAllReactionsFromUser($userId)
+    {
+        return new ApiResponse($this->getService(ReactionManager::class)->getReactions('', [], $userId));
+    }
+
+    /**
+     * @Route("/api/user/{userId}/reactions/{id}", options={"acl":{"public"}})
+     */
+    public function getReactionsFromUser($userId, $id)
+    {
+        $id = array_map('trim', explode(',', $id));
+        return new ApiResponse($this->getService(ReactionManager::class)->getReactions('', $id, $userId));
+    }
+    /**
+     * @Route("/api/reactions/{idreaction}/{id}/{page}/{username}", methods={"DELETE"}, options={"acl":{"public", "+"}})
+     */
+    public function deleteReaction($idreaction, $id, $page, $username)
+    {
+        if ($user = $this->wiki->getUser()) {
+            if ($username == $user['name'] || $this->wiki->UserIsAdmin()) {
+                $this->getService(ReactionManager::class)->deleteUserReaction($page, $idreaction, $id, $username);
+                return new ApiResponse(
+                    [
+                        'idReaction'=>$idreaction,
+                        'id'=>$id,
+                        'page' => $page,
+                        'user'=> $username
+                    ],
+                    Response::HTTP_OK
+                );
+            } else {
+                return new ApiResponse(
+                    ['error' => 'Seul les admins ou l\'utilisateur concerné peuvent supprimer les réactions.'],
+                    Response::HTTP_UNAUTHORIZED
+                );
+            }
+        } else {
+            return new ApiResponse(
+                ['error' => 'Vous devez être connecté pour supprimer les réactions.'],
+                Response::HTTP_UNAUTHORIZED
+            );
+        }
+    }
+
+    /**
+     * @Route("/api/reactions", methods={"POST"}, options={"acl":{"public", "+"}})
+     */
+    public function addReactionFromUser()
+    {
+        if ($user = $this->wiki->getUser()) {
+            if ($_POST['username'] == $user['name'] || $this->wiki->UserIsAdmin()) {
+                if ($_POST['reactionid']) {
+                    if ($_POST['pagetag']) { // save the reaction
+                        //get reactions from user for this page
+                        $userReactions = $this->getService(ReactionManager::class)->getUserReactions($user['name'], $_POST['pagetag'], [$_POST['reactionid']]);
```

Comment:
Lors de mes tests, j'ai une erreur à cette ligne:
`Uncaught Error: Call to undefined method YesWiki\Core\Service\ReactionManager::getUserReactions() in C:\xampp2\user\doryphore\website\includes\controllers\ApiController.php:173`